Frameworks/CoreText/CTLine.mm

```diff
@@ -367,43 +353,30 @@ CFIndex CTLineGetStringIndexForPosition(CTLineRef lineRef, CGPoint position) {
 */
 CGFloat CTLineGetOffsetForStringIndex(CTLineRef lineRef, CFIndex charIndex, CGFloat* secondaryOffset) {
     CGFloat ret = 0.0;
-    if (secondaryOffset) {
-        *secondaryOffset = 0.0;
-    }
-
-    if (lineRef == nil || charIndex < 0) {
-        return ret;
-    }
-
-    // if charIndex is greater than the last string index, return line's width as offset
-    if (charIndex > (CTLineGetGlyphCount(lineRef) - 1)) {
-        ret = (static_cast<_CTLine*>(lineRef))->_width;
-        if (secondaryOffset) {
-            *secondaryOffset = ret;
-        }
-
-        return ret;
-    }
-
-    CFArrayRef glyphRuns = CTLineGetGlyphRuns(lineRef);
-    CFIndex numberOfRuns = CFArrayGetCount(glyphRuns);
-    CFIndex currentIndex = -1;
-
-    for (int i = 0; i < numberOfRuns; i++) {
-        CTRunRef currRun = (CTRunRef)CFArrayGetValueAtIndex(glyphRuns, i);
-        CFIndex glyphCount = CTRunGetGlyphCount(currRun);
-
-        if (currentIndex + glyphCount >= charIndex) {
-            ret = (static_cast<_CTRun*>(currRun))->_glyphOrigins[charIndex - currentIndex - 1].x;
-            if (secondaryOffset) {
-                *secondaryOffset = ret;
+    if (lineRef && charIndex >= 0) {
+        _CTLine* line = static_cast<_CTLine*>(lineRef);
+        if (charIndex > line->_strRange.location + line->_strRange.length) {
+            ret = line->_width;
```

Comment:
discussed offline, reference platform returns line width

---
In reply to: [79688032](https://github.com/Microsoft/WinObjC/pull/989#discussion_r79688032) [](ancestors = 79688032)